tests/ops/api/v1/test_middleware.py

```diff
@@ -0,0 +1,128 @@
+# pylint: disable=missing-docstring, redefined-outer-name
+import json
+from datetime import datetime
+from typing import List
+
+import pytest
+from fastapi_pagination import Params
+
+from fides.api.api.v1.scope_registry import USER_CREATE
+from fides.api.cryptography.cryptographic_util import str_to_b64_str
+from fides.api.cryptography.schemas.jwt import (
+    JWE_ISSUED_AT,
+    JWE_PAYLOAD_CLIENT_ID,
+    JWE_PAYLOAD_SCOPES,
+)
+from fides.api.models.client import ClientDetail
+from fides.api.models.fides_user import FidesUser
+from fides.api.oauth.jwt import generate_jwe
+from fides.core.config import CONFIG
+
+# from tests.conftest import generate_auth_header_for_user
+
+page_size = Params().size
+
+import json
+from typing import Any, Dict, Generator
+
+import pytest
+
+from fides.api import middleware as _middleware
+from fides.api.api.v1.scope_registry import USER_CREATE
+from fides.api.cryptography.cryptographic_util import str_to_b64_str
+from fides.api.cryptography.schemas.jwt import (
+    JWE_ISSUED_AT,
+    JWE_PAYLOAD_CLIENT_ID,
+    JWE_PAYLOAD_SCOPES,
+)
+from fides.api.oauth.jwt import generate_jwe
+from fides.core.config import CONFIG
+
+# from sqlalchemy.exc import SQLAlchemyError
+
+
+@pytest.fixture
+def test_audit_log_resource_data() -> Generator:
+    """
+    Returns a well-formed dictionary of test data
+    """
+    audit_log_resource_data = {
+        "user_id": "test_user_id",
+        "request_path": "some/path",
+        "request_type": "POST",
+        "fides_keys": ["new_fides_key"],
+        "extra_data": {"key": "value"},
+    }
+    yield audit_log_resource_data
+
+
+@pytest.fixture
+def test_bad_audit_log_resource_data() -> Generator:
+    """
+    Returns a well-formed dictionary of test data
+    """
+    audit_log_resource_data = {
+        "user_id": None,
+        "request_path": "some/path",
+        "request_type": "POST",
+        "fides_keys": ["new_fides_key"],
+    }
+    yield audit_log_resource_data
+
+
+def test_record_written_to_db(db, test_audit_log_resource_data: Dict[str, Any]) -> None:
+    assert _middleware.write_audit_log_resource_record(db, test_audit_log_resource_data)
```

Comment:
what are we actually asserting here? i also think this would need to be awaited since it's an async function right?

i may be wrong about this, just stuck out to me on first glance. anyway, we don't need to hold up the PR, just noting here so we don't lose track of this.

Reply:
resolving the conversation to unblock merging but let's come back to this unless i'm missing the point!